agrona/src/main/java/org/agrona/concurrent/ShutdownSignalBarrier.java

```diff
@@ -36,7 +36,11 @@
     {
         for (final String signalName : SIGNAL_NAMES)
         {
-            Signal.handle(new Signal(signalName), (signal) -> LATCHES.forEach(CountDownLatch::countDown));
+            final var previous = Signal.handle(new Signal(signalName), (signal) -> {});
+            Signal.handle(new Signal(signalName), (signal) -> {
+                LATCHES.forEach(CountDownLatch::countDown);
+                previous.handle(signal);
```

Comment:
`previous` might be `null`.

Reply:
Also the formatting is wrong.

Reply:
i can add a guard, if you look in Signal.handle i do not believe the method can return null.

Reply:
fixed the formatting and added the guard (to be safe either way). Appreciate the time reviewing this 😄

Reply:
Looking at the `java.lang.Terminator` it does setup the same `SignalHandler` for `INT/TERM/HUP` signals so in this particular case the `previous` will never be `null`.